### ISSUE TITLE
Write content for Cluster Architecture concept

### ISSUE THREAD

**Issue by abhineettikoo** [kubernetes/website#29049]
Can we have a cluster architecture diagram (Low-level architecture ) on this page?

**Comment by chrismetz09** [comment]
something like this? https://brennerm.github.io/posts/kubernetes-overview-diagrams.html

**Comment by abhineettikoo** [comment]
Yes, it is more descriptive. Thanks.

**Comment by sftim** [comment]
I'd like to see a change where we simplify [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) so that (for example) `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) becomes a one line summary that links to a longer explanation in [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/).

Along with that change, we simplify the cluster diagram in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) to show the control plane as a “black box”, _but_ with obvious signposting via hyperlinks to a more detailed diagram in [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/) that explains what is inside that control plane, along with the detailed text.

Even better:
- add more diagrams in https://kubernetes.io/docs/concepts/architecture/ - including a control plane running as static pods on specially marked nodes
- migrate detailed information in https://kubernetes.io/docs/concepts/architecture/nodes/ to the reference section, and add a diagram to the node concept that shows what runs inside a typical node. Have the cluster architecture page link to that diagram.

/triage accepted
/priority important-longterm
/language en

**Comment by sftim** [comment]
/retitle Write content for Cluster Architecture concept

**Comment by sftim** [comment]
Relevant to #29071 as well

**Comment by shannonxtreme** [comment]
I mocked up a diagram for the main Cluster Architecture page. It abstracts the control plane and nodes. Any suggestions for making it clearer in the diagram that you can see the individual concepts for more detailed diagrams? Or do we clarify that in the text? 

![sampleclusterarchitecture](https://user-images.githubusercontent.com/30392854/127021954-3eed06dd-954b-49b3-9223-2bd38b34707e.jpg)

(I have SVGs but GitHub said no)

/assign

**Comment by sftim** [comment]
Remember that the cloud provider API integration is optional; I wouldn't want to lose that detail.
In the key to the simplified diagram, I would want to make it clear that Nodes are the places where containers execute.

Can we draw a diagram around the nodes to show “cluster network”? Does that help?

**Comment by shannonxtreme** [comment]
@sftim like this?

![sampleclusterarchitecture](https://user-images.githubusercontent.com/30392854/127172232-ca879ce5-cf39-47f1-9633-9c919f4bf149.png)

And then for the control plane architecture, this?

![detailedclusterarchitecture](https://user-images.githubusercontent.com/30392854/127172306-33bc0119-8017-4ae4-9769-3568a062e32d.png)

**Comment by sftim** [comment]
Actually, for all these diagrams, it might be helpful to show the cloud provider's API as being outside the cluster.

**Comment by shannonxtreme** [comment]
Yeah I figured. I'll rework them :)

**Comment by shannonxtreme** [comment]
@sftim I added updated diagrams for the following: 

1. General cluster architecture
2. Detailed cluster architecture
3. Detailed node architecture

I'm not sure what a diagram for a control plane running as pods would look like but I'll do some digging.

**Comment by sftim** [comment]
Take a look at the broad strokes of how https://azure.microsoft.com/en-gb/overview/kubernetes-getting-started/ explains things. There's some good ideas there; obviously we don't want to breach Microsoft's copyright but the overall approach can inform our work.

**Comment by sftim** [comment]
> control plane running as pods

I think I would use the node architecture diagram, label it “control plane node”, and add some named static pods there. Not sure how we show they are static pods. Maybe a double border plus explanatory labels?

**Comment by sftim** [comment]
We should also add a diagram into https://kubernetes.io/docs/concepts/architecture/control-plane-node-communication/

**Comment by Ritikaa96** [comment]
> I'd like to see a change where we simplify [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) so that (for example) `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) becomes a one line summary that links to a longer explanation in [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/).

Hi @sftim going through the page [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) , if we want to write about `kube-controller-manager` in a summary only, there are two things to notice:

- More kubernetes components mentioned there have to be summarised just the same.
- [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/) doesn't have explanatory information about `kube-controller-manager` so we have to add it in the page
- Where can we find information regarding `kube-controller-manager`:
   - A mention in https://kubernetes.io/docs/concepts/architecture/controller/
   - `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/)

In short wouldn't it be better to just add the current lengthy information for `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) to `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) instead of [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/)?

What do you think?

**Comment by sftim** [comment]
> In short wouldn't it be better to just add the current lengthy information for `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) to `kube-controller-manager` in [Kubernetes Components](https://kubernetes.io/docs/concepts/overview/components/) instead of [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/)?

There are lots of different routes to the same goal. Anything that moves us closer to our desired state is useful.
Long term, I think https://kubernetes.io/docs/concepts/overview/components/ should have _less_ detail, with some of what's there now moving to https://kubernetes.io/docs/concepts/architecture/control-plane/

Does that align with what you were thinking @abhineettikoo ?

**Comment by sftim** [comment]
Pull requests to make changes for this issue are welcome. If you can make the PR _small_ - even if the work doesn't feel completely finished - that's even better.

**Comment by sftim** [comment]
There's a diagram now!
